shared/src/main/scala/tastyquery/ast/Trees.scala

```diff
@@ -259,22 +275,34 @@ object Trees {
   /**  @param meth   A reference to the method.
     *  @param tpt    Not an EmptyTree only if the lambda's type is a SAMtype rather than a function type.
     */
-  case class Lambda(meth: Tree, tpt: TypeTree) extends Tree
+  case class Lambda(meth: Tree, tpt: TypeTree) extends Tree {
+    override def calculateType: Type =
+      tpt.toType
```

Comment:
this isn't right because `tpt` is only populated for SAM types, here is what dotty does:
```scala
def assignType(tree: untpd.Closure, meth: Tree, target: Tree)(using Context): Closure =
    tree.withType(
      if (target.isEmpty) meth.tpe.widen.toFunctionType(isJava = meth.symbol.is(JavaDefined), tree.env.length)
      else target.tpe)
```

Reply:
`toFunctionType` is fairly complex

Reply:
Looks like this is another case where we will need to be able to resolve `meth` to a `MethodType`, so I stubbed it out for now.